Link Eval refs to Eval feature (#789)
spiceai/docs: website/docs/reference/spicepod/evals.md

 - `json_match`: Checks for an equivalent JSON between expected and actual outputs.
 - `includes`: Checks for the actual output to include the expected output.
 - `fuzzy_match`: Checks whether a normalised version (ignoring casing, punctuation, articles (e.g. a, the), excess whitespace) of either the expected and actual outputs are a subset of the other.

> For an overview of what Evals are, and how to run them in Spice, see the [Evals](/docs/features/large-language-models/evals) documentation.

Reviewer: ```suggestion
To learn about Evals, including what they are and how to run them in Spice, refer to the [Evals documentation](/docs/features/large-language-models/evals).
```
